src/openms/source/ANALYSIS/ID/AhoCorasickDA.cpp

```diff
@@ -0,0 +1,1154 @@
+    auto new_size = old_size + 256;
+    // Exception size of DA becomes to large to store
+    if (new_size >= 1<<22)
+    {
+      throw Exception::InvalidSize(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, new_size);
+    }
+    // Add new empty nodes
+    bc_.resize(new_size);
+    // Set build informations and linkage
+    for (auto i = old_size; i < new_size; ++i)
+    {
+      build_info_.emplace_back(i);
+    }
+    // Add the new nodes to linkage
+    if (first_empty_elem_ == 0)
+    {
+      build_info_[old_size].prev = new_size - 1;
+      build_info_[new_size - 1].next = old_size;
+      first_empty_elem_ = old_size;
+    }
+    else
+    {
+      auto emp_tail = build_info_[first_empty_elem_].prev;
+      build_info_[old_size].prev = emp_tail;
+      build_info_[emp_tail].next = old_size;
+      build_info_[first_empty_elem_].prev = new_size - 1;
+      build_info_.back().next = first_empty_elem_;
+    }
+  }
+  void AhoCorasickDA::buildTrie_(Size begin, const Size end, const Size depth, const UInt32 node_pos)
+  {
+    // Reaches end of a string
+    if (sequences_[unique_idx_[begin]].size() == depth)
+    {
+      bc_[node_pos].term_flag = 1;
+      output_[node_pos] = unique_idx_[begin];
+      ++begin;
+      // String is not substring of another string
+      if (begin == end)
+      {
+        bc_[node_pos].base = 0;
+        bc_[node_pos].leaf_flag = 1;
+        return;
+      }
+    }
+    // Only one string in subtrie -> store in tail
+    if (begin+1 == end)
+    {
+      bc_[node_pos].leaf_flag = 1;
+      bc_[node_pos].base = static_cast<UInt32>(tail_.size());
+      const String& seq = sequences_[unique_idx_[begin]];
+      appendToTail_(&seq[depth]);
+      output_[-(tail_.size()-1)] = unique_idx_[begin];
+      return;
+    }
+    // Creates vector containing all transition labels
+    edges_.clear();
+    char current_label = sequences_[unique_idx_[begin]][depth];
+    for (auto i = begin; i < end; ++i)
+    {
+      char next_label = sequences_[unique_idx_[i]][depth];
+      if (current_label != next_label)
+      {
+        if (isAmbiguous_(current_label))
+        {
+          throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Input peptide must NOT contain ambiguous amino acids (B/J/Z/X) or characters other than amino acids!");
+        }
+        edges_.push_back(current_label);
+        current_label = next_label;
+      }
+    }
+    if (isAmbiguous_(current_label))
+    {
+      throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Input peptide must NOT contain ambiguous amino acids (B/J/Z/X) or characters other than amino acids!");
+    }
+    edges_.push_back(current_label);
+    // Return a base value which is valid for all labels in edges
+    auto base_value = xCheck_();
+
+    // Set values
+    bc_[node_pos].base = base_value;
+    build_info_[base_value].base_flag = 1;
+    for (auto const label : edges_)
```

Comment:
write `const char` instead of `auto` to make clear what happens here.
`auto` suggests some complicated type...